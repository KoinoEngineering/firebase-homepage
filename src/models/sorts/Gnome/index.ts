import Comparsion, { ComparsionParams } from "../Comparsion";

interface GnomeAdditionalProperty {
  cursor: number;
}
type GnomeParams = ComparsionParams & Partial<GnomeAdditionalProperty>;
export default class Gnome
    extends Comparsion
    implements GnomeAdditionalProperty
{
  public name = "ノームソート";
  public cursor: number;
  constructor({ items, compareCount, swapCount, cursor }: GnomeParams) {
      super({ items, compareCount, swapCount });
      this.cursor = cursor || 0;
  }

  public compare = async (): Promise<Gnome> => {
      const { items, compareCount, cursor } = this;
      if (!this.ended()) {
          return new Gnome({
              ...this,
              items: items.map((item, i) => {
                  return {
                      ...item,
                      comparing: i === cursor || i === cursor + 1,
                  };
              }),
              compareCount: compareCount + 1,
          });
      }
      return this;
  };

  public swap = async () => {
      if (!this.ended()) {
          const { items, cursor, swapCount } = this;
          // 比較中のものを取り出す
          const comparings = items.filter((item) => item.comparing);

          //   とりあえず2個じゃなかったら落とす
          if (comparings.length !== 2) {
              throw new Error("The item being compared must be 2 items");
          }

          // 比較中のものの順序をそろえる
          let isSwapped = false;
          const swapped =
        comparings[0].value > comparings[1].value
            ? ((isSwapped = true), [comparings[1], comparings[0]])
            : comparings;

          //   並べ替える
          let mapCursor = 0;
          const sorted = items.map((item, i) => {
              if (item.comparing) {
                  const swapTarget = swapped[mapCursor];
                  const fixed =
            i === items.length - 1 ||
            ((cursor === 0 || !isSwapped) && mapCursor === 0)
                ? true
                : swapTarget.fixed;
                  mapCursor++;
                  return {
                      ...swapTarget,
                      fixed,
                  };
              } else {
                  return item;
              }
          });
          return new Gnome({
              ...this,
              items: sorted,
              cursor:
          !isSwapped || cursor === 0
              ? sorted.findIndex((i) => !i.fixed)
              : cursor - 1,
              swapCount: isSwapped ? swapCount + 1 : swapCount,
          });
      }
      return this;
  };
}
