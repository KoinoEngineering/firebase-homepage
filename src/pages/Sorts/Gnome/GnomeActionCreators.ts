import { ActionType, ChangeValueAction, CursorNextAction, CursorPrevAction, EndAction, InitAction, StartAction, SwapAction } from "./GnomeActions";

export default {
    changeValue: (payload: ChangeValueAction["payload"]): ChangeValueAction => ({ type: ActionType.CHANGE_VALUE, payload }),
    cursorPrev: (): CursorPrevAction => ({ type: ActionType.CURSOR_PREV }),
    cursorNext: (): CursorNextAction => ({ type: ActionType.CURSOR_NEXT }),
    start: (): StartAction => ({ type: ActionType.START, payload: { running: true } }),
    init: (): InitAction => ({ type: ActionType.INIT }),
    swap: (): SwapAction => ({ type: ActionType.SWAP }),
    end: (): EndAction => ({ type: ActionType.END, payload: { running: false } }),
};