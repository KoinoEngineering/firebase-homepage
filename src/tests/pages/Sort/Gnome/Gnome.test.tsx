import React from "react";
import Gnome from "src/pages/Sorts/Gnome/Gnome";
import { renderWithProvider } from "src/tests/tetutils";

test("Gnome expect to match snapshot", () => {
    const target = renderWithProvider(<Gnome />);
    expect(target).toMatchSnapshot();
});
