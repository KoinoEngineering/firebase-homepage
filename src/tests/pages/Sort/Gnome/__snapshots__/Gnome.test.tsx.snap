// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gnome expect to match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root GnomePageContainer-root-3 MuiContainer-maxWidthLg"
        id="Gnome"
      >
        <div
          class="MuiGrid-root MuiGrid-container GridRow-container-12 GridRow-container-14"
          id="ContentsArea"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2"
          >
            ノームソート
          </h2>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container GridRow-container-12 GridRow-container-149 MuiGrid-spacing-xs-4"
          id="SettingArea"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-true"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined Mui-disabled Mui-disabled MuiFormLabel-filled"
                data-shrink="true"
              >
                要素数
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                  disabled=""
                  type="text"
                  value="20"
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-marginDense"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M4 12c0 .55.45 1 1 1h6c.55 0 1-.45 1-1s-.45-1-1-1H5c-.55 0-1 .45-1 1zm15 6h-2V7.38L14 8.4V6.7L18.7 5h.3v13z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    class="MuiButtonBase-root MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M9 7c-.55 0-1 .45-1 1v3H5c-.55 0-1 .45-1 1s.45 1 1 1h3v3c0 .55.45 1 1 1s1-.45 1-1v-3h3c.55 0 1-.45 1-1s-.45-1-1-1h-3V8c0-.55-.45-1-1-1zm11 11h-2V7.38L15 8.4V6.7L19.7 5h.3v13z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-236 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-238 PrivateNotchedOutline-legendNotched-239"
                  >
                    <span>
                      要素数
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-true"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root makeStyles-root-240 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                data-shrink="true"
              >
                向き
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              >
                <div
                  aria-haspopup="listbox"
                  class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                  role="button"
                  tabindex="0"
                >
                  昇順
                </div>
                <input
                  type="hidden"
                  value="asc"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-236 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-238 PrivateNotchedOutline-legendNotched-239"
                  >
                    <span>
                      向き
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-true"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined Mui-disabled Mui-disabled MuiFormLabel-filled"
                data-shrink="true"
              >
                処理間隔
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                  disabled=""
                  type="text"
                  value="0"
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-marginDense"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M4 12c0 .55.45 1 1 1h6c.55 0 1-.45 1-1s-.45-1-1-1H5c-.55 0-1 .45-1 1zm15 6h-2V7.38L14 8.4V6.7L18.7 5h.3v13z"
                        />
                      </svg>
                    </span>
                  </button>
                  <button
                    class="MuiButtonBase-root MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M9 7c-.55 0-1 .45-1 1v3H5c-.55 0-1 .45-1 1s.45 1 1 1h3v3c0 .55.45 1 1 1s1-.45 1-1v-3h3c.55 0 1-.45 1-1s-.45-1-1-1h-3V8c0-.55-.45-1-1-1zm11 11h-2V7.38L15 8.4V6.7L19.7 5h.3v13z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-236 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-238 PrivateNotchedOutline-legendNotched-239"
                  >
                    <span>
                      処理間隔
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container GridRow-container-12 GridRow-container-256"
          id="ControlArea"
        >
          <div
            class="MuiGrid-root MuiGrid-container GridRow-container-12 GridRow-container-258"
            id="Reset"
          >
            <h3
              class="MuiTypography-root MuiTypography-h3"
            >
              リセット
            </h3>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container GridRow-container-12 GridRow-container-260 MuiGrid-spacing-xs-4"
            id="Reset"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  昇順
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  降順
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  ランダム
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container GridRow-container-12 GridRow-container-291"
            id="ControlArea"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-true"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  開始
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container GridRow-container-12 GridRow-container-293"
          id="ContentsArea"
        >
          <div
            class="MuiGrid-root makeStyles-root-1 MuiGrid-container"
            id="SquareContainer"
          >
            <div
              class="MuiGrid-root makeStyles-root-294 MuiGrid-container MuiGrid-align-items-xs-flex-end"
              id="ReplaceSortContents"
            >
              <div
                class="MuiGrid-root makeStyles-root-295 makeStyles-root-297"
              >
                100
              </div>
              <div
                class="MuiGrid-root makeStyles-root-295 makeStyles-root-299"
              >
                95
              </div>
              <div
                class="MuiGrid-root makeStyles-root-295 makeStyles-root-301"
              >
                90
              </div>
              <div
                class="MuiGrid-root makeStyles-root-295 makeStyles-root-303"
              >
                85
              </div>
              <div
                class="MuiGrid-root makeStyles-root-295 makeStyles-root-305"
              >
                80
              </div>
              <div
                class="MuiGrid-root makeStyles-root-295 makeStyles-root-307"
              >
                75
              </div>
              <div
                class="MuiGrid-root makeStyles-root-295 makeStyles-root-309"
              >
                70
              </div>
              <div
                class="MuiGrid-root makeStyles-root-295 makeStyles-root-311"
              >
                65
              </div>
              <div
                class="MuiGrid-root makeStyles-root-295 makeStyles-root-313"
              >
                60
              </div>
              <div
                class="MuiGrid-root makeStyles-root-295 makeStyles-root-315"
              >
                55
              </div>
              <div
                class="MuiGrid-root makeStyles-root-295 makeStyles-root-317"
              >
                50
              </div>
              <div
                class="MuiGrid-root makeStyles-root-295 makeStyles-root-319"
              >
                45
              </div>
              <div
                class="MuiGrid-root makeStyles-root-295 makeStyles-root-321"
              >
                40
              </div>
              <div
                class="MuiGrid-root makeStyles-root-295 makeStyles-root-323"
              >
                35
              </div>
              <div
                class="MuiGrid-root makeStyles-root-295 makeStyles-root-325"
              >
                30
              </div>
              <div
                class="MuiGrid-root makeStyles-root-295 makeStyles-root-327"
              >
                25
              </div>
              <div
                class="MuiGrid-root makeStyles-root-295 makeStyles-root-329"
              >
                20
              </div>
              <div
                class="MuiGrid-root makeStyles-root-295 makeStyles-root-331"
              >
                15
              </div>
              <div
                class="MuiGrid-root makeStyles-root-295 makeStyles-root-333"
              >
                10
              </div>
              <div
                class="MuiGrid-root makeStyles-root-295 makeStyles-root-335"
              >
                5
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root GnomePageContainer-root-3 MuiContainer-maxWidthLg"
      id="Gnome"
    >
      <div
        class="MuiGrid-root MuiGrid-container GridRow-container-12 GridRow-container-14"
        id="ContentsArea"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2"
        >
          ノームソート
        </h2>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container GridRow-container-12 GridRow-container-149 MuiGrid-spacing-xs-4"
        id="SettingArea"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-true"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined Mui-disabled Mui-disabled MuiFormLabel-filled"
              data-shrink="true"
            >
              要素数
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled=""
                type="text"
                value="20"
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-marginDense"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M4 12c0 .55.45 1 1 1h6c.55 0 1-.45 1-1s-.45-1-1-1H5c-.55 0-1 .45-1 1zm15 6h-2V7.38L14 8.4V6.7L18.7 5h.3v13z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M9 7c-.55 0-1 .45-1 1v3H5c-.55 0-1 .45-1 1s.45 1 1 1h3v3c0 .55.45 1 1 1s1-.45 1-1v-3h3c.55 0 1-.45 1-1s-.45-1-1-1h-3V8c0-.55-.45-1-1-1zm11 11h-2V7.38L15 8.4V6.7L19.7 5h.3v13z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-236 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-238 PrivateNotchedOutline-legendNotched-239"
                >
                  <span>
                    要素数
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-true"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root makeStyles-root-240 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink="true"
            >
              向き
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            >
              <div
                aria-haspopup="listbox"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                role="button"
                tabindex="0"
              >
                昇順
              </div>
              <input
                type="hidden"
                value="asc"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-236 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-238 PrivateNotchedOutline-legendNotched-239"
                >
                  <span>
                    向き
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-true"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined Mui-disabled Mui-disabled MuiFormLabel-filled"
              data-shrink="true"
            >
              処理間隔
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled=""
                type="text"
                value="0"
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-marginDense"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M4 12c0 .55.45 1 1 1h6c.55 0 1-.45 1-1s-.45-1-1-1H5c-.55 0-1 .45-1 1zm15 6h-2V7.38L14 8.4V6.7L18.7 5h.3v13z"
                      />
                    </svg>
                  </span>
                </button>
                <button
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M9 7c-.55 0-1 .45-1 1v3H5c-.55 0-1 .45-1 1s.45 1 1 1h3v3c0 .55.45 1 1 1s1-.45 1-1v-3h3c.55 0 1-.45 1-1s-.45-1-1-1h-3V8c0-.55-.45-1-1-1zm11 11h-2V7.38L15 8.4V6.7L19.7 5h.3v13z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-236 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-238 PrivateNotchedOutline-legendNotched-239"
                >
                  <span>
                    処理間隔
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container GridRow-container-12 GridRow-container-256"
        id="ControlArea"
      >
        <div
          class="MuiGrid-root MuiGrid-container GridRow-container-12 GridRow-container-258"
          id="Reset"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3"
          >
            リセット
          </h3>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container GridRow-container-12 GridRow-container-260 MuiGrid-spacing-xs-4"
          id="Reset"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                昇順
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                降順
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                ランダム
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container GridRow-container-12 GridRow-container-291"
          id="ControlArea"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-true"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                開始
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container GridRow-container-12 GridRow-container-293"
        id="ContentsArea"
      >
        <div
          class="MuiGrid-root makeStyles-root-1 MuiGrid-container"
          id="SquareContainer"
        >
          <div
            class="MuiGrid-root makeStyles-root-294 MuiGrid-container MuiGrid-align-items-xs-flex-end"
            id="ReplaceSortContents"
          >
            <div
              class="MuiGrid-root makeStyles-root-295 makeStyles-root-297"
            >
              100
            </div>
            <div
              class="MuiGrid-root makeStyles-root-295 makeStyles-root-299"
            >
              95
            </div>
            <div
              class="MuiGrid-root makeStyles-root-295 makeStyles-root-301"
            >
              90
            </div>
            <div
              class="MuiGrid-root makeStyles-root-295 makeStyles-root-303"
            >
              85
            </div>
            <div
              class="MuiGrid-root makeStyles-root-295 makeStyles-root-305"
            >
              80
            </div>
            <div
              class="MuiGrid-root makeStyles-root-295 makeStyles-root-307"
            >
              75
            </div>
            <div
              class="MuiGrid-root makeStyles-root-295 makeStyles-root-309"
            >
              70
            </div>
            <div
              class="MuiGrid-root makeStyles-root-295 makeStyles-root-311"
            >
              65
            </div>
            <div
              class="MuiGrid-root makeStyles-root-295 makeStyles-root-313"
            >
              60
            </div>
            <div
              class="MuiGrid-root makeStyles-root-295 makeStyles-root-315"
            >
              55
            </div>
            <div
              class="MuiGrid-root makeStyles-root-295 makeStyles-root-317"
            >
              50
            </div>
            <div
              class="MuiGrid-root makeStyles-root-295 makeStyles-root-319"
            >
              45
            </div>
            <div
              class="MuiGrid-root makeStyles-root-295 makeStyles-root-321"
            >
              40
            </div>
            <div
              class="MuiGrid-root makeStyles-root-295 makeStyles-root-323"
            >
              35
            </div>
            <div
              class="MuiGrid-root makeStyles-root-295 makeStyles-root-325"
            >
              30
            </div>
            <div
              class="MuiGrid-root makeStyles-root-295 makeStyles-root-327"
            >
              25
            </div>
            <div
              class="MuiGrid-root makeStyles-root-295 makeStyles-root-329"
            >
              20
            </div>
            <div
              class="MuiGrid-root makeStyles-root-295 makeStyles-root-331"
            >
              15
            </div>
            <div
              class="MuiGrid-root makeStyles-root-295 makeStyles-root-333"
            >
              10
            </div>
            <div
              class="MuiGrid-root makeStyles-root-295 makeStyles-root-335"
            >
              5
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
